commit d7e4cce76f2be7878db95db9e0c18f0e433ba67e (refs/changes/72/5172/2)
Author: Josh Wilsdon <josh@wilsdon.ca>
Date:   2018-12-03T10:36:31-08:00 (10 months ago)
    
    TRITON-994 CNAPI crashes trying to talk to CNs with no cn-agent
    Reviewed by: Orlando Vazquez <orlando@joyent.com>
    Approved by: Orlando Vazquez <orlando@joyent.com>

diff --git a/lib/endpoints/vms.js b/lib/endpoints/vms.js
index f56651f..7d2ec1c 100644
--- a/lib/endpoints/vms.js
+++ b/lib/endpoints/vms.js
@@ -49,6 +49,9 @@ function errDetails(error) {
     var details;
 
     details = error.cause() && error.cause().body && error.cause().body.details;
+    if (!details) {
+        details = {};
+    }
 
     return (details);
 }
diff --git a/package.json b/package.json
index 69889d0..10bcb5a 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "cnapi",
   "description": "SmartDataCenter Compute Node API",
-  "version": "1.17.0",
+  "version": "1.17.1",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
